components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -310,21 +310,22 @@ public String streamingPredict(int tenantId, String userName, long modelId, Stri
                 unencodedData.add(item.clone());
             }
             List<?> predictions = predict(tenantId, userName, modelId, data);
-            String predictionsWithData = new String();
+            StringBuilder predictionsWithData = new StringBuilder();
             for (int i = 0; i < predictions.size(); i++) {
-                predictionsWithData += Arrays.toString(unencodedData.get(i)).replaceAll(
-                        MLConstants.WHITE_SPACE_SQUARE_BRACKET_REGEX, "")
-                        + csvFormat.getDelimiter() + String.valueOf(predictions.get(i)) + MLConstants.NEW_LINE;
+                predictionsWithData.append(MLUtils.arrayToCsvString(unencodedData.get(i), csvFormat.getDelimiter())
+                        + String.valueOf(predictions.get(i)) + MLConstants.NEW_LINE);
```

Comment:
String concatenations inside predictionsWithData.append(), using '+', can be replaced with append()

predictionsWithData.append(MLUtils.arrayToCsvString(unencodedData.get(i), csvFormat.getDelimiter()))
                                                .append(String.valueOf(predictions.get(i))).append(MLConstants.NEW_LINE);